src/thermo/SurfPhase.cpp

```diff
@@ -26,6 +26,7 @@ SurfPhase::SurfPhase(const std::string& infile, const std::string& id_) :
     m_press(OneAtm)
 {
     setNDim(2);
+    assignDensity(INFINITY);
```

Comment:
While I acknowledge that the PR is still in draft mode, I have a question about the introduction of C++11 `INFINITY` (as defined [here](https://en.cppreference.com/w/cpp/numeric/math/INFINITY)). 

Are there additional ramifications about this choice when it comes to Cantera/enhancements#125? If I understand correctly, there won't be an issue as long as we don't check for equality, but I'm not sure what the compiler warnings for `INFINITY` correspond to in actual usage. As I was hoping for a resolution of the `--fast-math` issue for 3.0, I wanted to get a better understanding.

Reply:
Great question!

Reply:
Linking https://stackoverflow.com/questions/22931147/stdisinf-does-not-work-with-ffast-math-how-to-check-for-infinity

Reply:
I'm not interested in pursuing support of `--ffast-math`, and have already explained why I think we shouldn't hamstring ourselves with such a limitation in the issues linked in https://github.com/Cantera/enhancements/issues/125.

Reply:
Would it make any difference if volumetric density were (arbitrarily) defined as zero instead in this case? While I see the point of not necessarily wanting to enable a controversial optimization approach, I’d still be in favor of not introducing roadblocks when not necessary.

Reply:
It is true that zero wouldn't be consistent with the original intent stated at the top, i.e.

> * Define the "volume" of a `SurfPhase` to be zero. This implies that the density must be `inf`.

However: as the concept of a volume evaluated for a 2D geometry is fundamentally flawed I am starting to wonder whether this shouldn't be better handled as an exception?

Reply:
> However: as the concept of a volume evaluated for a 2D geometry is fundamentally flawed I am starting to wonder whether this shouldn't be better handled as an exception?

To avoid losing any resulting discussion, can I propose that this question and any answers move to the main thread, out of this line comment?